### DISCUSSION TITLE
1.37 - better failure reporting edition

### DISCUSSION THREAD

**Discussion by cerebrate** [arkane-systems/genie#144] [General]
* Reduced default systemd startup timeout to 60s.
 * Added display of failed systemd units on timeout.

<hr /><em>This discussion was created from the release <a href='https://github.com/arkane-systems/genie/releases/tag/v1.37'>1.37 - better failure reporting edition</a>.</em>

**Comment by PavelSosin-320** [comment]
The catch is "no single log collection point exists". 
The distro is started implicitly. .wslconfig contains logfile parameter but that tells Kernel boot story but the file is huge and hardly readable. 
wsl --list shows distro's status. If failed wsl shows undocumented return code.
Then, systemctl is-system-running shows aggregated system status, systemctl --failed shows failed units and if journald has started journalctl -b tells the rest of the story.

**Comment by PavelSosin-320** [comment]
After a recent dev-pre-release of Windows that brought the new Linux distro path (wsl.local) and WSL Kernel 5.10 it doesn't work. Installation doesn't work. I'm trying manually to put things to the right places in my Fedora34. The pre-release of the official Fedora 34 is expected tomorrow. 🤣
The root cause is the incorrect placement of genie executable files in /usr/libexec/ . This path is not in the $PATH when WSL distro starts. The /etc/environment processing is skipped by WSL init program, i.e. there is no way to create a global environment for all users, root and rootless. It is impossible to add libexec to the PATH for the root user because in the .bashrc or .profile because bash and templates of these files distinguish between login via pam and non-login modes. Most of the session configuration is skipped in non-login mode when bash is not invoked from logind.
The Kernel 5.10 is much bigger than 5.4, it takes a much longer time to start. All boot stages are executed in chaotic order.
Meanwhile, I copied genie to /bin and everything works again.  Why not install genie in the /bin/ if all systemd binaries are installed there or create a link /bin/genie to /usr/libexec/genie/genie?

**Reply by cerebrate** [reply]
All the packages, that for Fedora included, already include a /usr/bin/genie -> /usr/libexec/genie/genie symlink.
